src/classes/presentation_controller.vala

```
@@ -753,6 +753,41 @@ namespace pdfpc {
             queue_pointer_surface_draws();
         }
 
+        /**
+         * Executes the script specified with --external-script.
+         * The script is called with the following parameters:
+         * - name of pdf file
+         * - total slide count
+         * - current slide number
+         * - current user slide number
+         *
+         * If the script exits with a non-zero return value, whatever
+         * the script wrote to stdout is printed in the console.
+         */ 
+        public void execute_external_script() {
+            if (Options.external_script == "none") {
```

I would silently ignore the "execute" request in that case.